app/controllers/admin/routes.py

```diff
@@ -387,16 +393,32 @@ def manageServiceLearningCourses(term=None):
     manageTerm = Term.get_or_none(Term.id == term) or g.current_term
 
     setRedirectTarget(request.full_path)
-
-    return render_template('/admin/manageServiceLearningFaculty.html',
-                            courseInstructors = getInstructorCourses(),
-                            unapprovedCourses = unapprovedCourses(term),
-                            approvedCourses = approvedCourses(term),
-                            terms = selectSurroundingTerms(g.current_term),
-                            term = manageTerm,
-                            cpPreview= session.get('cpPreview',{}),
-                            cpPreviewErrors = session.get('cpErrors',[])
-                           )
+    courseID = session.get("alterCourseId")
+    print(courseID)
+    if courseID:
+        session.pop("alterCourseId")
```

Comment:
Here is a good place to add a comment, I personally am not familiar with session variables so explaining what you are doing here could be good.

Reply:
Fixed